pkg/ingester/stream_rate_calculator.go

```diff
@@ -68,14 +73,15 @@ func (c *StreamRateCalculator) updateRates() {
 	for i := 0; i < c.size; i++ {
 		c.locks[i].Lock()
 		c.rates[i] = c.samples[i]
-		c.samples[i] = logproto.StreamRate{}
+		c.samples[i] = make(map[string]logproto.StreamRate)
 
 		sr := c.rates[i]
-		if sr.StreamHash != 0 {
+		for _, v := range sr {
 			rates = append(rates, logproto.StreamRate{
```

Comment:
not anymore. If there isn't anything here, the map of `tenant -> StreamRate` will just be empty because of how we reset things here https://github.com/grafana/loki/blob/8836d0a305a3ceec6aae8efdf7f053adc2661a4e/pkg/ingester/stream_rate_calculator.go#L76